src/picongpu/include/particles/interpolationMemoryPolicy/ShiftToValidRange.hpp

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2016 Richard Pausch
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+
+/** functor for particle field interpolator
+ *
+ * This functor is a simplification of the full
+ * field to particle interpolator that can be used in the
+ * particle pusher
+ */
+
+namespace picongpu
+{
+
+namespace interpolationMemoryPolicy
+{
+/** Shift memory and position to valid range
+ *  repositions memory and position if particle moves
+ *  outside of cell in sub-stepping cycle
+ *  returns shifted position and memory */
```

Comment:
Please add a some more information to this comment: valid range is if the position for each dimension is in the range `[0.0;1.0)`.